paperqa/contrib/openreview_paper_helper.py

```diff
@@ -0,0 +1,165 @@
+            }
+            submission_info_string += f"{paper}\n"
+        return submission_info_string
+
+    def fetch_relevant_papers(self, question: str) -> dict[str, Any]:
+        """Get relevant papers for a given question using LLM."""
+        submissions = self.get_submissions()
+        submission_string = self.create_submission_string(submissions)
+
+        if len(submission_string) > self.settings.parsing.chunk_size:
+            chunks = [
+                submission_string[i : i + self.settings.parsing.chunk_size]
+                for i in range(
+                    0, len(submission_string), self.settings.parsing.chunk_size
+                )
+            ]
+        else:
+            chunks = [submission_string]
+        relevant_papers = []
+        for chunk in chunks:
+            logger.info(f"Fetching relevant papers for question: {question}")
+            relevant_papers += self._get_relevant_papers_chunk(question, chunk)
+        subs = [s for s in submissions if s.id in set(relevant_papers)]
+        self.download_papers(subs)
+        return {sub.id: sub for sub in subs}
+
+    def _get_relevant_papers_chunk(self, question: str, chunk: str) -> list[Any]:
+        prompt = (
+            chunk
+            + "You are the helper model that aims to get up to 20 most relevant papers for the user's question. "
+            + "User's question:\n"
+        )
+
+        response = completion(
+            model=self.settings.llm,
+            messages=[{"role": "user", "content": prompt + question}],
+            response_format={
+                "type": "json_object",
+                "response_schema": RELEVANT_PAPERS_SCHEMA,
+            },
+            verbose=self.settings.verbosity,
+        )
+
+        content = json.loads(response.choices[0].message.content)
+        return [p["submission_id"] for p in content["suggested_papers"]]
+
+    def download_papers(self, submissions: list[Any]) -> None:
+        """Download PDFs for given submissions."""
+        downloaded_papers = Path(self.settings.paper_directory).rglob("*.pdf")
+        downloaded_ids = [p.stem for p in downloaded_papers]
+        logger.info("Downloading PDFs for relevant papers.")
+        for submission in submissions:
+            if submission.id not in downloaded_ids:
+                self._download_pdf(submission)
+
+    def _download_pdf(self, submission: Any) -> bool:
+        """Download a single PDF."""
+        pdf_link = f"https://openreview.net/{submission.content['pdf']['value']}"
+        response = httpx.get(pdf_link)
+        SUCCESS_CODE = 200
+        if response.status_code == SUCCESS_CODE:
+            with open(
+                f"{self.settings.paper_directory}/{submission.id}.pdf", "wb"
+            ) as f:
+                f.write(response.content)
+            return True
+        logger.warning(
+            f"Failed to download the PDF. Status code: {response.status_code}"
+        )
```

Comment:
Would you mind putting `response.text` here, so people can see the failure reason